cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/managers/TransactionManager.java

```
@@ -58,24 +57,56 @@ public static TransactionManager getInstance(CvaConfiguration cvaConfiguration)
+                            EvidenceSetDBAdaptor.QueryParams.REPORTED_VARIANTS.key(),
+                            IdentifierBuilder.buildIdentifier(reportedVariant)
+                    );
+                    List<EvidenceSet> evidenceSets = evidenceSetManager.find(queryByReportedVariantId).getResult();
+                    if (evidenceSets != null) {
+                        // 3. if any remove reference and update
+                        for (EvidenceSet evidenceSet : evidenceSets) {
+                            evidenceSetManager.removeReportedVariant(evidenceSet, reportedVariant);
+                        }
+                    }
+                }
+            }
+            // remove reported variants
             reportedVariantManager.removeByTransaction(transactionId);
+            // remove observed variants
             observedVariantManager.removeByTransaction(transactionId);
+            // remove variants
             variantManager.removeByTransaction(transactionId);
-            evidenceSetManager.removeByTransaction(transactionId);
             // Update transaction status as CANCELLED (Indicates rollback operation completed)
             logger.info(String.format("Start updating transaction [%s] to CANCELLED status", transactionId));
             update(transactionId, TransactionStatus.CANCELLED, errorMessage); // Transaction status
             // Note: Failures in rollback operation are handled by a background job
         } catch (Exception exception) {
-            logger.error("Failure in rollback operation for transaction [" + transactionId + "] ");
+            String message = String.format(
+                    "Failure in rollback operation for transaction [%s]: %s",
+                    transactionId,
+                    exception.getMessage()
+            );
+            logger.error(message);
+            update(transactionId, TransactionStatus.ROLLBACK_ERROR, message);
             throw new CvaException(exception.getMessage());
```

Should we throw this exception now?